server/src/test/java/org/elasticsearch/action/admin/cluster/hotthreads/NodesHotThreadsRequestTests.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.hotthreads;
+
+import org.elasticsearch.action.admin.cluster.node.hotthreads.NodesHotThreadsRequest;
+import org.elasticsearch.action.support.nodes.BaseNodesRequest;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.monitor.jvm.HotThreads;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class NodesHotThreadsRequestTests extends ESTestCase {
+
+    /** Simple override of BaseNodesRequest to ensure we read the
+     * common fields of the nodes request.
+     */
+    static class NodesHotThreadsRequestHelper extends BaseNodesRequest<NodesHotThreadsRequestHelper> {
+        NodesHotThreadsRequestHelper(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        NodesHotThreadsRequestHelper(String... nodesIds) {
+            super(nodesIds);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+        }
+    }
+
+    public void testBWCSerialization() throws IOException {
+        BytesStreamOutput out = new BytesStreamOutput();
+
+        TimeValue sampleInterval = new TimeValue(50, TimeUnit.MINUTES);
+
+        NodesHotThreadsRequestHelper outHelper = new NodesHotThreadsRequestHelper("123");
+
+        outHelper.writeTo(out);
+        // Write manually some values that differ from the defaults
+        // in NodesHotThreadsRequest
+        out.writeInt(4); // threads
```

Comment:
Instead of manually creating a stream, you can set the version on the stream and check round trip. This is more robust as future changes may be made to the format. There are helpers to get a random version constant to test with, see `VersionUtils`.

Reply:
Thanks for this! I'll look into this and make the change with my follow-up PR in the same area.